streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -332,7 +363,7 @@ public void remove() {
         }
 
         public void close() {
-            openIterators.remove(this);
+            callback.deregisterIterator(this);
```

Comment:
If the object is created with the no-argument constructor, this will throw an NPE.

Reply:
What's the benefits of using a callback here than calling `openIterators` directly? I think adding to `openIterators` outside of the constructor makes sense, but cannot think of the rationale of doing this upon closure.

Reply:
I felt it was appropriate to make the iterator class(es) static, in which case we need a way to remove from openIterators on closure. Will fix NPE when I add emptyIterator() method

Reply:
Ack, thanks for the explanation.